elastic/kibana
# [Fleet] Fix enrollment flyout with fleet server from policy page

## Summary

Resolve #104474
Resolve #104524

When opening the "add agent" instructions from the policy details page we were not showing the fleet server instructions if the policy contains the fleet server instructions.

This was caused because we were only checking for the presence of fleet server in the agent policy selection step and we are not using that step if a policy is already selected. I moved the check of the presence of the fleet server at the top level of the component so we check it every time.

Also when opening the add agent flyout we now show the instruction to add fleet server if it's not set even if fleet server host is not set)

The agent flyout should show the following:

With no selected policies:
* no fleet server => add a fleet server ...
* Fleet server but no fleet server host => the fleet server host settings callout 
* Fleet server host and fleet server => managed instructions

With a selected policy
* no fleet server => add a fleet server ...
* no fleet server host and the selected policy is fleet server => add a fleet server
* no fleet server host and the selected policy is not fleet server => the fleet server host settings callout 


## How to tests

Go on on the details page for a policy with Fleet server you should see the instructions to setup Fleet server

<img width="1452" alt="Screen Shot 2021-07-06 at 4 40 10 PM" src="https://user-images.githubusercontent.com/1336873/124664082-eb4bfe80-de78-11eb-8f97-82ef0d4a4284.png">
